### PR TITLE
fix: subpixel calculations for table separators

### PR DESCRIPTION
**Pull request by 13ANM**
[FX-2260]
[SPT-2114](https://toptal-core.atlassian.net/browse/SPT-2114)

### This is a **POC**, open discussion to possible alternatives (current PR implementation is not definitive)

Chromium based browsers have a different rendering engine that handles the subpixel calculations different than other browsers, meaning they don't really support subpixel rendering and this leads to integer approximations on random elements that, in our case for the table rows borders, will become inconsistent.

For some users, depending on their monitor this will also happen without needing to zoom into the page, reason for which we should try and find a workaround for this until chromium will accept subpixel rendering.

With this approach we can add the borders to the table cells that will adjust based on the adjacent borders, so at least now, when they adjust while zooming, they should adjust all at once

### How to test

1. Go to [deploy](https://picasso.toptal.net/SPT-2114-inconsistent-table-separators/?path=/story/components-table--table) on a chromium based browser (Chrome, Brave, Edge, Vivaldi etc.)
2. Use all zoom variants provided by the browser and check if there are any discrepancies between the table separators
3. Compare to [current](https://picasso.toptal.net/?path=/story/components-table--table) as in the video below

https://user-images.githubusercontent.com/79158829/141304105-3fd779fe-01fd-417e-8064-0613db4aa8e4.mov

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

[FX-2260]: https://toptal-core.atlassian.net/browse/FX-2260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
For me, it works differently than for you. The separator is the same for every row, with your fix it just holds the height of the separator, so it doesn't seem so bugy as for you :)
image
Left is yours fix, right is master.

I don't mind solution

**Comment by 13ANM** [comment]
@TomasSlama Yes, this issue also depends on the screen and OS that you're using I guess, we can push this forward only if it fixes it for everyone and there are no downsides, as it was only reported by a few users

**Comment by sergiubutnarasu** [comment]
I have the same behavior as you in the staff portal, but in the Picasso storybook, I have the same behavior as @TomasSlama mentioned.

For me, your solution is working.

**Comment by OndrejTuma** [comment]
I'm just wondering if the border on table row should be removed then 🤔 
Also some snapshots needs updating, maybe use something like `yarn test:unit --updateSnapshot`?

File: packages/picasso/src/TableCell/styles.ts [hunk 1]
```diff
@@ -33,11 +33,14 @@ const narrowCellStyles = {
...
 }
 
-export default ({ palette, typography }: Theme) =>
+export default ({ sizes, palette, typography }: Theme) =>
   createStyles({
     root: regularCellStyles,
     compact: compactCellStyles,
     narrow: narrowCellStyles,
+    bordered: {
+      borderBottom: `${sizes.borderWidth} solid ${palette.grey.lighter2}`
```
Comment (denieler) [inline packages/picasso/src/TableCell/styles.ts:42]: can you please explain why do we need the bottom border and on TableCell and on TableRow?

Comment (13ANM) [reply packages/picasso/src/TableCell/styles.ts:42]: @denieler I've removed it from the `TableRow`, the idea is to move the border to the cells, so in these cases when we zoom they will calculate differently, based on the adjacent borders

**Comment by 13ANM** [comment]
As discussed, we can abandon this for now since it affects only a few users and wait for a `chromium` or `material-ui` update

**Pull request closed without merging**